File: LeetCode/Python/Delete Characters to Make Fancy String.py
"""
  Delete Characters to Make Fancy String
  URL: https://leetcode.com/problems/delete-characters-to-make-fancy-string
  Difficulty: Easy
  Topic : String
"""

class Solution:
    def makeFancyString(self, s: str) -> str:
        char = " "
        count = 0
        result = ""
        for c in s:
            if c == char:
                if count < 2:
                    result += c
                count += 1
            else:
                char = c
                count = 1
                result += c
        return result

File: LeetCode/Easy/Delete Characters to Make Fancy String.swift
/*
  Delete Characters to Make Fancy String
  URL: https://leetcode.com/problems/delete-characters-to-make-fancy-string
  Difficulty: Easy
  Topic : String
*/

class Solution {
    func makeFancyString(_ s: String) -> String {
        s.reduce(into: (Character(" "),0,"")) { p, c in 
            if c == p.0 {
                if p.1 < 2 {
                    p.2 += "\(c)"
                }
                p.1 += 1
            } else {
                p.0 = c
                p.1 = 1
                p.2 += "\(c)"
            }
        }.2
    }
}
